### ISSUE TITLE
$response->getErrors() empty when there are errors

### ISSUE THREAD

**Issue by patrickmj** [omeka/omeka-s#89]
Mentioned this to @jimsafley . When doing `$api->create`, I'm getting a response back that has errors, but the `$response->getErrors()` array is empty.

Tried using batchCreate(), but same result. Either way, I figure we should be able to use create() and get more about the errors that happen?

Here's what does show up when running the import script via the command line

`

Fatal error: Uncaught exception 'Doctrine\ORM\ORMException' with message 'The EntityManager is closed.' in /var/www/omekas/vendor/doctrine/orm/lib/Doctrine/ORM/ORMException.php on line 144

Doctrine\ORM\ORMException: The EntityManager is closed. in /var/www/omekas/vendor/doctrine/orm/lib/Doctrine/ORM/ORMException.php on line 144
`

**Comment by jimsafley** [comment]
I suspect that an error isn't set to the response because the exception is somehow thrown from outside the API context, so the manager isn't catching it. As to the underlying problem, this error typically happens when an exception occurs during flush() (therefore a DBAL exception), which rolls back the transaction and closes the entity manager. I think any subsequent call to flush() will raise this error.

I'd enable SQL logging and see if there's anything unusual.

**Comment by patrickmj** [comment]
Found the problem in a variable I was missetting, which made the SQL fail. Should SQL errors show up as errors on the response?

**Comment by jimsafley** [comment]
The API manager catches _all_ exceptions that occur within the API context. Somehow this is happening outside of it, so there's no way to set the error to the API response.

**Comment by zerocrates** [comment]
If you had or had access to the trace that would tell us where it's throwing at

**Comment by patrickmj** [comment]
Should something like this produce something in  `$response->getErrors()`

```
2015-07-25T14:09:43-04:00 ERR (3): exception 'Omeka\Api\Exception\BadRequestException' with message 'The API does not support the "omeka2items" resource.' in /var/www/omekas/application/src/Api/Manager.php:157
Stack trace:
#0 /var/www/omekas/application/src/Api/Manager.php(86): Omeka\Api\Manager->execute(Object(Omeka\Api\Request))
#1 /var/www/omekas/modules/Omeka2Importer/src/Job/Import.php(157): Omeka\Api\Manager->read('omeka2items', 50)
#2 /var/www/omekas/modules/Omeka2Importer/src/Job/Import.php(47): Omeka2Importer\Job\Import->importedItem(50)
#3 /var/www/omekas/modules/Omeka2Importer/src/Job/Import.php(34): Omeka2Importer\Job\Import->importItems()
#4 /var/www/omekas/application/src/Job/Strategy/SynchronousStrategy.php(26): Omeka2Importer\Job\Import->perform()
#5 /var/www/omekas/application/src/Job/Dispatcher.php(82): Omeka\Job\Strategy\SynchronousStrategy->send(Object(Omeka\Entity\Job))
#6 /var/www/omekas/data/scripts/perform-job.php(45): Omeka\Job\Dispatcher->send(Object(Omeka\Entity\Job), Object(Omeka\Job\Strategy\SynchronousStrategy))
#7 {main}
2015-07-25T14:09:43-04:00 ERR (3): exception 'Omeka\Api\Exception\BadRequestException' with message 'The API does not support the "omeka2items" resource.' in /var/www/omekas/application/src/Api/Manager.php:157
Stack trace:
#0 /var/www/omekas/application/src/Api/Manager.php(52): Omeka\Api\Manager->execute(Object(Omeka\Api\Request))
#1 /var/www/omekas/modules/Omeka2Importer/src/Job/Import.php(62): Omeka\Api\Manager->create('omeka2items', Array)
#2 /var/www/omekas/modules/Omeka2Importer/src/Job/Import.php(34): Omeka2Importer\Job\Import->importItems()
#3 /var/www/omekas/application/src/Job/Strategy/SynchronousStrategy.php(26): Omeka2Importer\Job\Import->perform()
#4 /var/www/omekas/application/src/Job/Dispatcher.php(82): Omeka\Job\Strategy\SynchronousStrategy->send(Object(Omeka\Entity\Job))
#5 /var/www/omekas/data/scripts/perform-job.php(45): Omeka\Job\Dispatcher->send(Object(Omeka\Entity\Job), Object(Omeka\Job\Strategy\SynchronousStrategy))
#6 {main}
```


The line producing it is

`
$response = $this->api->create('omeka2items', $importItemEntityJson);
`

Granted, there's something for me to sort out with the configuration for the manager, but this seems like something that should come back from getErrors(), or at least not just silently continue without indication of problem.

**Comment by jimsafley** [comment]
When the API manager catches an exception it sets it to the response object. Interestingly, we don't add the exception message to the response's error store. Maybe we should. It would be a simple matter of adding 
```php
$response->addError('exception', $e->getMessage());
```
to the last catch in `Manager::execute()`.

**Comment by zerocrates** [comment]
Yeah, errors and exceptions get some different treatment because we use exceptions to figure out if we should produce a "not found" or various other kinds of errors.

The trick here is that both the controller code that uses the API and the JSON API renderer look for an exception and rethrow or print info about it, so those "normal" channels wouldn't have failed silently. Saving the exception like that gets a little weird though, so we should probably have a way to make it just throw normally to avoid this kind of problem where we've got other things using the API.

It's entirely possible that these should just be regular throws instead of saving on the Response. We've changed this around a few times already and it gets simpler each time.
